### ISSUE TITLE
Block raw input from controller

### ISSUE THREAD

**Issue by SiXoS** [x360ce/x360ce#288]
I'm using a logitec dual action controller when I'm playing star wars: Battlefront. The problem is that the game is compatible with any controller but you cannot specify the key bindings. Before I used x360ce everything was on the wrong button. When I use x360ce the bindings are correct but the game registers both the raw input from the controller and the xbox 360 input so it becomes sort of a race of which input will be used. I don't know if it is possible but could you block the raw input from being registered by the game?

**Comment by Nucleoprotein** [comment]
I think you need correct HookMask, try to add HookDI.

**Comment by SiXoS** [comment]
@tapcio Will try that.
To clarify: this is the battlefront released yesterday in 2015.

**Comment by SiXoS** [comment]
I tried at first with just DI but it didn't worked so I marked all and tested again. No change there is still an input race.

**Comment by Squall-Leonhart** [comment]
probably needs HookMask = 0x0000000E

**Comment by SiXoS** [comment]
Nope, HookMask COM, DI and PIDVID doesn't help. I should mention that when I press Apply/Synchronize Settings it gets an UnauthorizedException when writing to xinput1_3.dll. The next time I start x360ce it says that it has found unnecesary files and deletes xinput1_3.dll. When I start up the game the controller doesn't work at all.

**Comment by Estecka** [comment]
Any recent update on this matter ?

I'm trying to play Haven with a NS3137. None of the masks above help blocking the original input.

**Comment by Squall-Leonhart** [comment]
you can't block raw input.

**Comment by Nucleoprotein** [comment]
ViGEmBus version does not resolve this? 
@Squall-Leonhart 
https://docs.microsoft.com/en-us/windows/win32/inputdev/using-raw-input
Detouring and blocking (ie. always failing) RegisterRawInputDevices should disable RAWINPUT

**Comment by Squall-Leonhart** [comment]
VIGEM doesn't block the original controller without Cerberus/Guardian and there are still issues with it, particularly for unity3d stuff.

**Comment by EvoJocys** [comment]
This report is about the 3.x line, where x360ce was an xinput DLL copied into each game's folder. The current 4.x line works differently: it creates a virtual Xbox 360 controller that every game sees, with nothing copied into game folders. Game-specific problems from the DLL era do not carry over.

The latest release is 4.21 (September 2026): https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
